src/components/hv-text-field/index.tsx

```diff
@@ -36,47 +34,62 @@ const HvTextField = (props: HvComponentProps) => {
   // Extract known attributes into their own variables
   const autoFocus = props.element.getAttribute('auto-focus') === 'true';
   const debounceTimeMs =
-    parseInt(props.element.getAttribute('debounce'), 10) || 0;
-  const defaultValue = props.element.getAttribute('value');
+    Dom.safeParseIntAttribute(props.element, 'debounce') ?? 0;
+  const defaultValue = props.element.getAttribute('value') || undefined;
   const editable = props.element.getAttribute('editable') !== 'false';
-  const keyboardType = props.element.getAttribute('keyboard-type') || undefined;
+  const keyboardType =
+    (props.element.getAttribute('keyboard-type') as KeyboardTypeOptions) ||
+    undefined;
   const multiline =
     props.element.localName === LOCAL_NAME.TEXT_AREA ||
     props.element.getAttribute('multiline') === 'true';
   const secureTextEntry = props.element.getAttribute('secure-text') === 'true';
   const textContentType =
-    props.element.getAttribute('text-content-type') || 'none';
+    (props.element.getAttribute('text-content-type') as TextContextType) ||
+    'none';
 
   // Handlers
   const setFocus = (focused: boolean) => {
-    const newElement = props.element.cloneNode(true);
-    props.onUpdate(null, 'swap', props.element, { newElement });
+    const newElement = props.element.cloneNode(true) as Element;
+    if (props.onUpdate !== null) {
+      props.onUpdate(null, 'swap', props.element, { newElement });
 
-    if (focused) {
-      Behaviors.trigger('focus', newElement, props.onUpdate);
-    } else {
-      Behaviors.trigger('blur', newElement, props.onUpdate);
+      if (focused) {
+        Behaviors.trigger('focus', newElement, props.onUpdate);
+      } else {
+        Behaviors.trigger('blur', newElement, props.onUpdate);
+      }
     }
   };
 
+  // TODO: move this to top
   // Create a memoized, debounced function to trigger the "change" behavior
+  // eslint-disable-next-line
   const triggerChangeBehaviors = useCallback(
     debounce((newElement: Element) => {
-      Behaviors.trigger('change', newElement, props.onUpdate);
+      if (props.onUpdate !== null) {
+        Behaviors.trigger('change', newElement, props.onUpdate);
+      }
     }, debounceTimeMs),
     [],
   );
 
   // This handler takes care of handling the state, so it shouldn't be debounced
   const onChangeText = (value: string) => {
     const formattedValue = HvTextField.getFormattedValue(props.element, value);
-    const newElement = props.element.cloneNode(true);
+    const newElement = props.element.cloneNode(true) as Element;
     newElement.setAttribute('value', formattedValue);
-    props.onUpdate(null, 'swap', props.element, { newElement });
+    if (props.onUpdate !== null) {
+      props.onUpdate(null, 'swap', props.element, { newElement });
+    }
     triggerChangeBehaviors(newElement);
   };
 
-  const textInputRef: ElementRef<typeof TextInput> = useRef(null);
+  // TODO: move this to top
```

Comment:
Definitely need to move this up

Reply:
do you know why?

Reply:
We'll get inconsistent results if the hooks are called conditionally